core/node/consensus/src/storage/registry_contract/committee_extractor.rs

```diff
@@ -0,0 +1,136 @@
+use anyhow::Context;
+use zksync_concurrency::{ctx, ctx::Ctx, scope, time};
+use zksync_consensus_roles::{attester, validator};
+use zksync_dal::consensus_dal;
+use zksync_node_api_server::execution_sandbox::BlockArgs;
+
+use crate::storage::{
+    registry_contract::{abi::CommitteeAttester, vm_reader::VMReader},
+    ConnectionPool,
+};
+
+#[derive(Debug)]
+pub struct CommitteeExtractor {
+    pool: ConnectionPool,
+    reader: VMReader,
+}
+
+impl CommitteeExtractor {
+    pub fn new(pool: ConnectionPool, reader: VMReader) -> CommitteeExtractor {
+        Self { pool, reader }
+    }
+
+    pub async fn run(mut self, ctx: &Ctx) -> anyhow::Result<()> {
+        tracing::info!("Running zksync_node_consensus::storage::CommitteeExtractor");
+        let res = scope::run!(ctx, |ctx, s| async {
+            let mut next_batch = self.next_batch_to_extract_committee(ctx).await?;
+            loop {
+                self.wait_batch(ctx, next_batch).await?;
+                self.extract_batch_committee(ctx, next_batch).await?;
```

Comment:
So `wait_batch` calls [get_sealed_l1_batch_number](https://github.com/matter-labs/zksync-era/blob/3f2cac6f2a8b81738098eb26d270be8494d2faac/core/lib/dal/src/blocks_dal.rs#L101) which is the last batch in the table, but it's not necessarily execute yet, or is it? What would happen if the contract was called with a number that it doesn't have data for, would it fail, or return an empty result?